### PR TITLE
Update some python dependencies

### PR DESCRIPTION
**Pull request by cmaglie**
**Please check if the PR fulfills these requirements**

- [x] The PR has no duplicates (please search among the [Pull Requests](https://github.com/arduino/arduino-cli/pulls)
      before creating one)
- [x] The PR follows
      [our contributing guidelines](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#pull-requests)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] `UPGRADING.md` has been updated with a migration guide (for breaking changes)

**What kind of change does this PR introduce?**
Upgraded some python dependencies to achieve `urllib3 >= 1.26.5`, as asked by dependabot.

- **Does this PR introduce a breaking change, and is [titled accordingly](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#breaking)?**
No

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -27,6 +27,7 @@ mdx-truly-sane-lists = "^1.2"
 GitPython = "^3.1.20"
 mike = "^1.0.1"
 black = "^21.7b0"
+click = "8.0.4"
```
Comment (per1234) [inline pyproject.toml:30]: Why was this added? `click` is only a transitive dependency, so I don't see why it should be in `pyproject.toml`.

**Review by per1234 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (cmaglie) [reply pyproject.toml:30]: Our pinned version of `black` doesn't run with the latest `click`: https://github.com/psf/black/issues/2964
I tried to update black to the latest 22.3 as someone else suggested, but poetry doesn't seem to find it...

Comment (per1234) [reply pyproject.toml:30]: That is odd, it has been in use for the last month without problems in the projects which have Dependabot-maintained Python dependencies (e.g., https://github.com/arduino/arduino-lint/pull/365, https://github.com/arduino/library-registry/pull/1243, https://github.com/arduino/library-registry-submission-parser/pull/92, https://github.com/arduino/libraries-repository-engine/pull/113)

Comment (cmaglie) [reply pyproject.toml]: Ok, I think I found the reason, I did `poetry install` just after changing the dependency, but you must do `poetry update` before. Now everything seems ok... let's wait for the CI to run, and see.

**Review by per1234 — APPROVED** [review]
